eclair-core/src/main/scala/fr/acinq/eclair/wire/protocol/LightningMessageTypes.scala

```diff
@@ -385,10 +385,9 @@ case class ChannelUpdate(signature: ByteVector64,
                          htlcMinimumMsat: MilliSatoshi,
                          feeBaseMsat: MilliSatoshi,
                          feeProportionalMillionths: Long,
-                         htlcMaximumMsat: Option[MilliSatoshi],
+                         htlcMaximumMsat: MilliSatoshi,
                          tlvStream: TlvStream[ChannelUpdateTlv] = TlvStream.empty) extends RoutingMessage with AnnouncementMessage with HasTimestamp with HasChainHash {
-
-  def messageFlags: Byte = if (htlcMaximumMsat.isDefined) 1 else 0
+  def messageFlags: Byte = 1
```

Comment:
We need this to construct `ChannelDisabled` error messages, which should really only use `channelFlags`.

Reply:
Yes, it looks a bit weird here if it's just a constant, but if we reintroduce more message flags (such as in #2362) it makes a bit more sense. Let's see if #2362 is accepted at the spec level and we'll see.